lib/doublemap_api/configuration.rb

```diff
@@ -8,12 +8,18 @@ class Configuration
     attr_accessor :adapter
     # The output stream to which debug information should be written
     attr_accessor :debug_output
+    # Optional procs called when a corresponding model object is created,
+    # before it is returned to a caller.
+    attr_accessor :on_route, :on_stop, :on_vehicle
 
     # The defaults to use for any configuration options that are not provided
     DEFAULT_CONFIGURATION = {
       version: '3.2', # Taken from a comment on "http://bus.gocitybus.com/RouteMap/Index"
       adapter: :httparty,
-      debug_output: false
+      debug_output: false,
+      on_route: ->(_){},
```

Comment:
As the default, I think using named variables would be useful. For example, this could be `->(_route){}` instead of just `_`. Nitpick, but unnamed underscores always feel a bit odd.